### ISSUE TITLE
Kodi check crashes with unparseable json payload

### ISSUE THREAD

**Issue by indianahorst** [languitar/autosuspend#45]
> $ tail -n 50 /var/log/syslog
> Sep  7 10:32:47 localhost autosuspend[22676]: 2018-09-07 10:32:47,078 - root - INFO - Configuring check XIdleTime with class XIdleTime
> Sep  7 10:32:47 localhost autosuspend[22676]: 2018-09-07 10:32:47,078 - loop - INFO - Starting new check iteration
> Sep  7 10:32:47 localhost autosuspend[22676]: 2018-09-07 10:32:47,078 - loop - DEBUG - Executing check Ping
>Sep  7 10:33:03 localhost autosuspend[22676]: 2018-09-07 10:33:03,198 - loop - DEBUG - Executing check Smb
> Sep  7 10:33:03 localhost autosuspend[22676]: 2018-09-07 10:33:03,238 - loop - DEBUG - Executing check Processes
> Sep  7 10:33:03 localhost autosuspend[22676]: 2018-09-07 10:33:03,248 - loop - DEBUG - Executing check Mpd
> Sep  7 10:33:03 localhost autosuspend[22676]: 2018-09-07 10:33:03,251 - mpd.base - INFO - Calling MPD connect('localhost', 6600, timeout=None)
> Sep  7 10:33:03 localhost autosuspend[22676]: 2018-09-07 10:33:03,260 - loop - WARNING - Check Mpd[class=Mpd] failed. Ignoring...
> Sep  7 10:33:03 localhost autosuspend[22676]: Traceback (most recent call last):
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 168, in check
> Sep  7 10:33:03 localhost autosuspend[22676]:     state = self._get_state()
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 160, in _get_state
> Sep  7 10:33:03 localhost autosuspend[22676]:     client.connect(self._host, self._port)
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/lib/python3/dist-packages/mpd/base.py", line 641, in connect
> Sep  7 10:33:03 localhost autosuspend[22676]:     self._sock = self._connect_tcp(host, port)
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/lib/python3/dist-packages/mpd/base.py", line 604, in _connect_tcp
> Sep  7 10:33:03 localhost autosuspend[22676]:     raise err
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/lib/python3/dist-packages/mpd/base.py", line 597, in _connect_tcp
> Sep  7 10:33:03 localhost autosuspend[22676]:     sock.connect(sa)
> Sep  7 10:33:03 localhost autosuspend[22676]: ConnectionRefusedError: [Errno 111] Connection refused
> Sep  7 10:33:03 localhost autosuspend[22676]: During handling of the above exception, another exception occurred:
> Sep  7 10:33:03 localhost autosuspend[22676]: Traceback (most recent call last):
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 559, in execute_checks
> Sep  7 10:33:03 localhost autosuspend[22676]:     result = check.check()
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 177, in check
> Sep  7 10:33:03 localhost autosuspend[22676]:     raise TemporaryCheckError(error)
> Sep  7 10:33:03 localhost autosuspend[22676]: TemporaryCheckError: [Errno 111] Connection refused
> Sep  7 10:33:03 localhost autosuspend[22676]: 2018-09-07 10:33:03,260 - loop - DEBUG - Executing check Kodi
> Sep  7 10:33:03 localhost autosuspend[22676]: Traceback (most recent call last):
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 760, in <module>
> Sep  7 10:33:03 localhost autosuspend[22676]:     main()
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 756, in main
> Sep  7 10:33:03 localhost autosuspend[22676]:     run_for=args.run_for)
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 581, in loop
> Sep  7 10:33:03 localhost autosuspend[22676]:     matched = execute_checks(_checks, all_checks, logger)
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 559, in execute_checks
> Sep  7 10:33:03 localhost autosuspend[22676]:     result = check.check()
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/bin/autosuspend", line 203, in check
> Sep  7 10:33:03 localhost autosuspend[22676]:     '?request={"jsonrpc": "2.0", '
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/lib/python3/dist-packages/requests/models.py", line 892, in json
> Sep  7 10:33:03 localhost autosuspend[22676]:     return complexjson.loads(self.text, **kwargs)
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
> Sep  7 10:33:03 localhost autosuspend[22676]:     return _default_decoder.decode(s)
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
> Sep  7 10:33:03 localhost autosuspend[22676]:     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
> Sep  7 10:33:03 localhost autosuspend[22676]:   File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
> Sep  7 10:33:03 localhost autosuspend[22676]:     raise JSONDecodeError("Expecting value", s, err.value) from None
> Sep  7 10:33:03 localhost autosuspend[22676]: json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
> Sep  7 10:33:03 localhost systemd[1]: autosuspend.service: Main process exited, code=exited, status=1/FAILURE
> Sep  7 10:33:03 localhost systemd[1]: autosuspend.service: Failed with result 'exit-code'.

**Comment by languitar** [comment]
So, the first issue here is that your MPD server doesn't accept the connection:

> ConnectionRefusedError: [Errno 111] Connection refused

Does it really run on 'localhost', with port 6600? Can you connect to it?

However, what actually kills autosuspend is your configured Kodi check. The checks seems to receive an unparsable answer from Kodi. Could you please provide your config file (with password removed) to see how both checks are configured?

**Comment by indianahorst** [comment]
Yes, MPD runs on port 6600 and I can connect to it:

`
# user                          "mpd"
# group                          "nogroup"
# For network
bind_to_address         "127.0.0.1"
bind_to_address         "192.168.1.8"
port                            "6600"
`

The following is my /etc/autosuspend.conf. There are no passwords in it:

[autosuspend.conf.txt](https://github.com/languitar/autosuspend/files/2362770/autosuspend.conf.txt)

**Comment by languitar** [comment]
Could you please execute the following command on the affected system and show the result:
```
python3 -c "import requests; print(requests.get('http://localhost:8080/jsonrpc?request{"jsonrpc": "2.0", "id": 1, "method": "Player.GetActivePlayers"}').text)"
```

**Comment by indianahorst** [comment]
When I execute the command above, I get an empty line.

The access to Kodi (which is version 17.6) is secured with Username and Password. 
The Settings in Kodi (Settings - Services - Control)  are:  

Allow remote control via HTTP: yes
Port 8080
Username: kodi
Password: xxxxxxx
Enable SSL: no 
Web Interface: KODI web interface - Chorus2

**Comment by languitar** [comment]
Oh ok. Authentication wasn't supported in 1.0. It is, however, since 2.0. I will fix the crash, though.

**Comment by languitar** [comment]
Yes, there is none, because ubuntu only ships version 1 and in that version, authentication wasn't supported.

However, I just notice that I have missed the Kodi check from including the authentication support. I will prepare a new release with that:  #47.

Btw, I have no idea why the manpage looks so bad in the online version.

**Comment by languitar** [comment]
Things have been fixed and released as 2.0.2. Debian packaging is on its way. Please try again with that release, especially for authentication.
